### ISSUE TITLE
[bug] no-leaked-resize-observer incorrectly reports leak when unobserve is used

### ISSUE THREAD

**Issue by NotWoods** [Rel1cx/eslint-react#808]
### Describe the bug

web-api/no-leaked-resize-observer should not flag resize observers that are cleaned up properly. `observer.unobserve` is a valid cleanup method and the rule should allow its use.

### Reproduction

```tsx
  useEffect(() => {
    const scrollRoot = scrollRootRef.current;
    if (!scrollRoot) {
      return undefined;
    }

    const resizeObserver = new ResizeObserver(getAndSetScrollOffsets);
    resizeObserver.observe(scrollRoot);

    return () => {
      resizeObserver.unobserve(scrollRoot);
    };
  }, [elementRef, scrollRootRef]);
```

### Expected behavior

The above effect should not trigger lint errors

### Platform and versions

```Text
Node v20.16.0, eslint-react 1.14.0
```
